### PR TITLE
[SNMP] Traps server cleanup and modularization

### PR DESCRIPTION
**Pull request by dplepage-dd**
### What does this PR do?
This refactors the SNMP traps server into several different modules, in preparation for componentizing them. This is being done as a separate PR upon which #19452 will be rebased so that #19452 can just be about the actual componentization.

### Motivation
This is part of NDMII-287.

### Describe how to test/QA your changes
Behavior should not change; standard snmp traps QA should still work.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by AlexandreYang — APPROVED** [review]

File: pkg/snmp/traps/status/status.go [hunk 1]
```diff
@@ -54,8 +86,8 @@ func GetStatus() map[string]interface{} {
 	}
 	status["metrics"] = metrics
 
-	if startError != nil {
-		status["error"] = startError.Error()
-	}
+	// if startError != nil {
+	// 	status["error"] = startError.Error()
+	// }
```
Comment (AlexandreYang) [inline pkg/snmp/traps/status/status.go:91]: **❓ question:** ‏Why this is commented?

**Review by AlexandreYang — COMMENTED** [review]
LGTM overall.

Could you have a look at this question: https://github.com/DataDog/datadog-agent/pull/19785/files#r1341265788

**Review by pgimalac — APPROVED** [review]
LGTM for ASC files

File: pkg/snmp/traps/status/status.go (hunk 1 above)
Comment (dplepage-dd) [reply pkg/snmp/traps/status/status.go]: Oops, I meant to remove that entirely. Once this is componentized, the error is handled as part of the app startup, so we won't have access to it.
Removed.

**Pull request merged**